tests/reftests/reinstall.test

```diff
@@ -85,3 +85,88 @@ The following actions would be performed:
   - recompile b 1
 === install 1 package
   - install   c 1
+### :::::: Check that reinstalling a non-installed package is equivalent to installing it
+### :::::: TODO: This is currently broken
+### <pkg:d.1>
+opam-version: "2.0"
+### opam reinstall d --cudf=reinstall --show --yes
+d is not installed. Install it? [y/n] y
+The following actions would be performed:
+=== install 1 package
+  - install d 1
+### opam install d --cudf=install --show --yes
+The following actions would be performed:
+=== install 1 package
+  - install d 1
+### diff -u install-1.cudf reinstall-1.cudf
+--- install-1.cudf	2022-11-08 17:20:07.261292279 +0000
++++ reinstall-1.cudf	2022-11-08 17:20:07.250292200 +0000
+@@ -1,5 +1,5 @@
+ # Solver: builtin-mccs+glpk
+-# Criteria: -removed,-count[avoid-version,changed],-count[version-lag,request],-count[version-lag,changed],-count[missing-depexts,changed],-changed
++# Criteria: -changed,-count[avoid-version:,true],-count[version-lag:,false],-count[missing-depexts:,true]
+ preamble: 
+ property: opam-query: int = [0], missing-depexts: int = [0], avoid-version: int = [0], version-lag: nat = [0], pinned: bool = [false], installed-root: bool = [false], reinstall: bool = [false], opam-version: string, opam-name: string
+ 
+@@ -45,7 +45,6 @@
+ package: d
+ version: 1
+ conflicts: d
+-opam-query: 1
+ opam-name: d
+ opam-version: 1
+ 
+# Return code 1 #
+### :::::: Check again with one install and one non-installed
+### :::::: TODO: This is currently broken
+### opam reinstall d a --cudf=reinstall --show --yes
```

Comment:
```suggestion
### <pkg:a>
opam-version: "2.0"
### opam reinstall d a --cudf=reinstall --show --yes
```
to avoid having `b` and `c` installs and their extra output

Reply:
I think it's fine to have `b` and `c` around. It exhibit more behaviours and still tests the code-path i need here.